cmd/admin-obd.go

```diff
@@ -0,0 +1,399 @@
+	Name:   "obd",
+	Usage:  "run on-board diagnostics",
+	Action: mainAdminOBD,
+	Before: setGlobalsFromContext,
+	Flags:  append(adminOBDFlags, globalFlags...),
+	CustomHelpTemplate: `NAME:
+  {{.HelpName}} - {{.Usage}}
+
+USAGE:
+  {{.HelpName}} TARGET
+
+FLAGS:
+  {{range .VisibleFlags}}{{.}}
+  {{end}}
+EXAMPLES:
+  1. Get server information of the 'play' MinIO server.
+     {{.Prompt}} {{.HelpName}} play/
+`,
+}
+
+type clusterOBDStruct struct {
+	Status string         `json:"status,omitempty"`
+	Error  string         `json:"error,omitempty"`
+	Info   madmin.OBDInfo `json:"obdInfo,omitempty"`
+}
+
+func (u clusterOBDStruct) String() string {
+	u.Status = ""
+	data, err := json.Marshal(u)
+	if err != nil {
+		fatalIf(probe.NewError(err), "unable to marshal into JSON.")
+	}
+	return string(data)
+}
+
+// JSON jsonifies service status message.
+func (u clusterOBDStruct) JSON() string {
+	statusJSONBytes, e := json.MarshalIndent(u, " ", "    ")
+	fatalIf(probe.NewError(e), "Unable to marshal into JSON.")
+
+	return string(statusJSONBytes)
+}
+
+// checkAdminInfoSyntax - validate arguments passed by a user
+func checkAdminOBDSyntax(ctx *cli.Context) {
+	if len(ctx.Args()) == 0 || len(ctx.Args()) > 1 {
+		cli.ShowCommandHelpAndExit(ctx, "obd", 1) // last argument is exit code
+	}
+}
+
+//compress and tar obd output
+func tarGZ(c clusterOBDStruct, alias string) error {
+	filename := fmt.Sprintf("%s-obd_%s.json.gz", alias, time.Now().Format("20060102150405"))
+	f, err := os.OpenFile(filename, os.O_CREATE|os.O_RDWR, 0644)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	defer func() {
+		fmt.Println("OBD data saved to", filename)
+	}()
+
+	gzWriter := gzip.NewWriter(f)
+	defer gzWriter.Close()
+
+	enc := json.NewEncoder(gzWriter)
+	if err := enc.Encode(c); err != nil {
+		return err
+	}
+
+	warningMsgBoundary := "*********************************************************************************"
+	warning := warnText("                                   WARNING!!")
+	warningContents := infoText(`     ** THIS FILE MAY CONTAIN SENSITIVE INFORMATION ABOUT YOUR ENVIRONMENT ** 
+     ** PLEASE INSPECT CONTENTS BEFORE SHARING IT ON ANY PUBLIC FORUM **`)
+
+	warningMsgHeader := infoText(warningMsgBoundary)
+	warningMsgTrailer := infoText(warningMsgBoundary)
+	fmt.Printf("%s\n%s\n%s\n%s\n", warningMsgHeader, warning, warningContents, warningMsgTrailer)
+
+	return nil
+}
+
+func infoText(s string) string {
+	return console.Colorize("INFO", s)
+}
+func greenText(s string) string {
+	return console.Colorize("GREEN", s)
+}
+func warnText(s string) string {
+	return console.Colorize("WARN", s)
+}
+
+func mainAdminOBD(ctx *cli.Context) error {
+	checkAdminOBDSyntax(ctx)
+
+	// Get the alias parameter from cli
+	args := ctx.Args()
+	aliasedURL := args.Get(0)
+
+	// Create a new MinIO Admin Client
+	client, err := newAdminClient(aliasedURL)
+	fatalIf(err, "Unable to initialize admin connection.")
+
+	types := &options
+
+	dot := "●"
```

Comment:
I see that we have repeated definitions of this constant can we make it global and use it everywhere? 

```
git grep 'dot :'
cmd/admin-heal.go:      dot := console.Colorize("Dot", " ●  ")
cmd/admin-info.go:      dot := "●"
```

Reply:
Yeah, this should go into a constant